### PR TITLE
Cfto

### PR DESCRIPTION
**Pull request by sumitroy2011**
These PR contains the following changes
1) Introduction of CFTO scenarios for Ireland Site
2) Perfect Partners functionality on both UK & IE CFTO Site

Please Merge!!

### CODE REVIEW

File: common/cftoFunctions.js [hunk 1]
```diff
@@ -45,18 +48,44 @@ function chooseCollection() {
     browser.click('.collection--summary__btn');
 }
 
+function chooseCollectionIE() {
+    //Select store from the store list and confirm
+    browser.click('#store_4048');
```
Comment (SunilSyal) [inline common/cftoFunctions.js]: I think think this ID can be dynamic. Can we use something static?

File: common/cftoFunctions.js [hunk 2]
```diff
@@ -45,18 +48,44 @@ function chooseCollection() {
...
+    browser.click('.collection-continue .btn--primary');
+
+    //Click to confirm the choosen collection slot data & move-on to payment page
+    browser.click('.collection--summary__btn');
+}
+
 function cardDetailsLightBox() {
     browser.selectByValue('select[name="cardType"]', '11105_Mastercard');
-    browser.setValue('#account', '5105105105105100');
+    browser.setValue('#account', '5454609899026213');
```
Comment (SunilSyal) [inline common/cftoFunctions.js:49]: R these real card details? If yes, it could be a threat to the account.

File: spec/cftoie/_A_CFTO_IE.test.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        var ey = ex.splice(3);
+        var ez = ex.slice(-3, 1); //It shows the name of the first perfect partner in the carousel
+
+
+        browser.click('.add-to-bag__btn');
+        var ab = browser.getText('.product-item__link:nth-of-type(1)');
+        var bc = ex.slice(-3, 1); //It shows the name of the perfect partner product added to the basket
+
+        // This checks whether the product from perfect partners carousel gets added to the basket & its same as first item in carousel
+        if (ez == bc) {
```
Comment (SunilSyal) [inline spec/cftoie/_A_CFTO_IE.test.js]: IF ez is not equal to bc, is that also a scenario? How are we going to test that?

File: spec/cftouk/_A_CFTO_UK.test.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        var ey = ex.splice(3);
+        var ez = ex.slice(-3, 1); //It shows the name of the first perfect partner in the carousel
+
+
+        browser.click('.add-to-bag__btn');
+        var ab = browser.getText('.product-item__link:nth-of-type(1)');
+        var bc = ex.slice(-3, 1); //It shows the name of the perfect partner product added to the basket
+
+        // This checks whether the product from perfect partners carousel gets added to the basket & its same as first item in carousel
+        if (ez == bc) {
```
Comment (SunilSyal) [inline spec/cftouk/_A_CFTO_UK.test.js]: Same as earlier.

File: spec/cftouk/_A_CFTO_UK.test.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+        // Confirm the order by clicking on the place order button
+        browser.click('input[type="submit"]');
+
+        //Confirming the details on the order confirmation page
+        expect(browser.element('.order-confirm-msg__content .order-confirm-msg__keyword').getText()).to.contain('401');
+    });
+
+
+    it('Place an UK CFTO order with guest user', function() {
```
Comment (SunilSyal) [inline spec/cftouk/_A_CFTO_UK.test.js:53]: Seems similar test cases as that for IE. Make a common function for these test cases and reuse.

File: common/cftoFunctions.js (hunk 2 above)
Comment (sumitroy2011) [reply common/cftoFunctions.js:49]: This is a Test Card only. Actually the previous card was throwing frequent errors. So, switch it to a more reliable test card

File: spec/cftouk/_A_CFTO_UK.test.js (hunk 5 above)
Comment (sumitroy2011) [reply spec/cftouk/_A_CFTO_UK.test.js:53]: No, Sir guest user flow of IE is bit different than that of UK. In IE user cannot search a store. He directly lands to the store selection page. Hence, different functions is mentioned for UK & IE

File: spec/cftouk/_A_CFTO_UK.test.js (hunk 4 above)
Comment (sumitroy2011) [reply spec/cftouk/_A_CFTO_UK.test.js]: Same as above

File: common/cftoFunctions.js (hunk 1 above)
Comment (sumitroy2011) [reply common/cftoFunctions.js]: Sir, This ID is based on the Ireland Store ID. So, this ID is always going to be same for a respective store. So, i have used the ID of a "DUBLIN MARY ST" store which is 4048

**Pull request closed without merging**